extensions/cornerstone/src/services/SegmentationService/SegmentationService.ts

```diff
@@ -667,23 +674,32 @@ class SegmentationService extends PubSubService {
 
       const centerWorld = derivedVolume.imageData.indexToWorld([x, y, z]);
 
-      segmentationSchema.cachedStats = {
-        ...segmentationSchema.cachedStats,
+      segmentation.cachedStats = {
+        ...segmentation.cachedStats,
         segmentCenter: {
-          ...segmentationSchema.cachedStats.segmentCenter,
+          ...segmentation.cachedStats.segmentCenter,
           [segmentIndex]: {
             center: {
               image: [x, y, z],
               world: centerWorld,
             },
-            modifiedTime: Date.now(),
+            modifiedTime: segDisplaySet.SeriesDate,
           },
         },
       };
 
-      this._broadcastEvent(EVENTS.SEGMENT_PIXEL_DATA_CREATED, {
-        segmentIndex: Number(segmentIndex),
-        numSegments,
+      const numInitialized = Object.keys(
```

Comment:
This is a strange name for an array... I suspect you want to do a `.length` on the `Object.keys` result here instead of on line 697 below.